docs/java/features/odata/generate-client.md

```diff
@@ -0,0 +1,270 @@
+---
+id: generate-java-odata-vmd-v2-v4
+title: OData VDM Generator for Java
+hide_title: false
+hide_table_of_contents: false
+sidebar_label: Generate a VDM for OData
+description: This article describes how to leverage the OData Generator to obtain Java classes from a service definition. These classes can then be used to build type-safe OData requests.
+keywords:
+- sap
+- cloud
+- sdk
+- odata
+- java
+- VDM
+- generate
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+
+## Generate a Virtual Data Model with the OData Generator ##
+
+The OData Generator allows for generating Java classes from the metadata of an OData service. These classes which are refered to as Virtual Data Model (VDM) provide type-safe access to the service.
+
+In general there are three ways to use the generator: 
+- Via the dedicated maven plugin
+- Via the CLI
+- By instantiating and invoking it at runtime
+
+The maven plugin is usually the recommended way as it integrates nicely with most project setups and makes configuration easy. However, the other two approaches are available and all are documented below.
+
+For all three the required input is an `EDMX` file holding the service metadata.
+
+:::note
+Please be aware that V2 and V4 service definitions are not interchangeable. In particular, the service definition of a V2 service can not serve as input to the V4 generator.
```

Comment:
OData v2 instead of V2? (and V4). Also I would always write this in lowercase ;)

Reply:
But v4 to v2 is possible?

Reply:
No, I'll clarify.

Reply:
Let's be consistent then. Let's add `Odata vX`  to guidelines.

Reply:
Done